toolkit/tools/imagegen/configuration/networkconfig.go

```diff
@@ -0,0 +1,309 @@
+
+	if err = n.validateIPAddress(n.NetMask); err != nil {
+		return fmt.Errorf("invalid input for netmask: %w", err)
+	}
+
+	if err = n.validateIPAddress(n.GateWay); err != nil {
+		return fmt.Errorf("invalid input for gateway: %w", err)
+	}
+
+	for _, nameserver := range n.NameServers {
+		if err = n.validateIPAddress(nameserver); err != nil {
+			return fmt.Errorf("invalid input for nameserver: %w", err)
+		}
+	}
+
+	return
+}
+
+// deviceIsValid returns an error if the Device name is empty
+func (n *Network) deviceIsValid() (err error) {
+	n.Device = strings.TrimSpace(n.Device)
+	if n.Device == "" {
+		return fmt.Errorf("invalid input for device, device cannot be empty")
+	}
+	return
+}
+
+func findBootIfValue() (deviceAddr string, err error) {
+	const macAddressStartIndex = 3
+
+	bootifValue, ferr := GetKernelCmdLineValue("BOOTIF")
+	bootifValue = strings.TrimSpace(bootifValue)
+	if ferr != nil || bootifValue == "" {
+		err = ferr
+		return
+	}
+
+	// The bootif value in the cmdline set by pxelinux is of the following format:
+	// bootif=01-MAC Address, where each byte value of the MAC address is separated
+	// by dashes instead of colons. Therefore, we're reading from the 10th spot of the
+	// string to obtain the MAC address and then replace the dashes with colons
+	deviceAddr = strings.Replace(bootifValue[macAddressStartIndex:len(bootifValue)], "-", ":", -1)
+	return
+}
+
+func checkNetworkDeviceAvailability(networkData Network) (deviceName string, err error) {
+	ifaces, err := net.Interfaces()
+	if err != nil {
+		return
+	}
+
+	if networkData.Device == "bootif" {
+		networkData.Device, err = findBootIfValue()
+		if err != nil {
+			logger.Log.Errorf("Failed to read bootif value from /proc/cmdline")
+			return
+		}
+	}
+
+	for _, iface := range ifaces {
+		if networkData.Device == iface.Name {
+			deviceName = networkData.Device
+			return
+		} else {
+			ifaceAddr := strings.TrimSpace(iface.HardwareAddr.String())
+
+			if strings.Contains(networkData.Device, ":") {
+				if ifaceAddr != "" && strings.EqualFold(ifaceAddr, networkData.Device) {
+					deviceName = iface.Name
+					return
+				}
+			}
+		}
+	}
+
+	return
+}
+
+func populateMatchSection(networkData Network, fileName, deviceName string) (err error) {
+	matchSection := fmt.Sprintf("[Match]\nName=%s\n", deviceName)
+	err = file.Append(matchSection+"\n", fileName)
+	if err != nil {
+		logger.Log.Errorf("Failed to write [Match] section: %s", err)
+	}
+
+	return
+}
+
+func populateNetworkSection(networkData Network, fileName string) (err error) {
+	var networkSection strings.Builder
+
+	// Write network section tag
+	networkSection.WriteString("[Network]\n")
+
+	// Currently only supports static IP setting
+	// Write IP and netmask
+	if networkData.NetMask != "" && networkData.Ip != "" {
+		stringMask := net.IPMask(net.ParseIP(networkData.NetMask).To4())
+		cidrPrefix, _ := stringMask.Size()
+		networkSection.WriteString(fmt.Sprintf("Address=%s/%d\n", networkData.Ip, cidrPrefix))
```

Comment:
What does "cidr" stand for?

Reply:
It stands for classless inter-domain routing. It's the number after the "/" and follows the IP address. Here's a more detailed explanation for reference: https://www.techtarget.com/searchnetworking/definition/CIDR#:~:text=CIDR%20(Classless%20Inter%2DDomain%20Routing)%20%2D%2D%20also%20known%20as,B%20and%20Class%20C%20networks.

Reply:
Thanks. If I'm getting this right, CIDR notation is the combination of the IP address followed by a `/` and a subnetwork mask (be that in a dotted-decimal or the more common single number form). Since you only have the last part of it here, I think simply calling it `mask`, `subnetMask`, `networkMask`, etc. might be clearer for the broader audience, while still being correct.